### PR TITLE
feat(subagent): add worktree isolation for filesystem-safe subagents

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Add `isolated=True` parameter to `subagent()` that runs the subagent in a git worktree, preventing it from modifying the parent's working directory.

- New module `gptme/util/git_worktree.py` with create/cleanup/get_git_root utilities
- `isolated=True` creates a worktree under `/tmp/gptme-worktrees/`; falls back to temp dir if not in a git repo
- Automatic cleanup after subagent completion (both thread and subprocess modes)
- 7 new tests covering creation, cleanup, isolation, and edge cases
- All 32 tests passing (7 new + 25 existing profile tests)

## Motivation

When multiple subagents run in parallel and modify files, they can conflict with each other and with the parent agent. Git worktree isolation (inspired by Claude Code's `isolation: "worktree"` pattern) gives each subagent its own copy of the repository.

Addresses item 2 of #1457.

## Test plan

- [x] 7 new tests for git worktree utilities (create, cleanup, auto-branch, isolation)
- [x] All 25 existing profile tests still pass
- [x] mypy clean on changed files
- [x] ruff clean

----

> [!IMPORTANT]
> Add worktree isolation for subagents with `isolated=True` parameter in `subagent()`, utilizing new git worktree utilities and tests.
> 
>   - **Behavior**:
>     - Add `isolated=True` parameter to `subagent()` in `subagent.py` to run subagents in a git worktree for filesystem isolation.
>     - Automatically cleans up worktree after subagent completion.
>   - **Utilities**:
>     - New module `git_worktree.py` with `create_worktree()`, `cleanup_worktree()`, and `get_git_root()` functions for managing git worktrees.
>   - **Tests**:
>     - Add 7 new tests in `test_git_worktree.py` for worktree creation, cleanup, and isolation.
>     - Ensure all 32 tests pass (7 new + 25 existing).
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=gptme%2Fgptme&utm_source=github&utm_medium=referral)<sup> for 775e78e7cd4d03e23514722dc7852e6878e633e5. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/util/git_worktree.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    if branch_name is None:
+        branch_name = f"subagent-{uuid.uuid4().hex[:8]}"
+
+    worktree_path = worktree_base / branch_name
+    worktree_base.mkdir(parents=True, exist_ok=True)
+
+    # Create worktree with a new branch based on HEAD
+    subprocess.run(
+        ["git", "worktree", "add", str(worktree_path), "-b", branch_name],
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/util/git_worktree.py:74]: Good catch. Fixed in 477d8e03 — switched from `random_string(4)` (36^4 ≈ 1.7M combinations) to `uuid.uuid4().hex[:8]` (16^8 ≈ 4B combinations), so the collision probability is negligible even with many concurrent subagents sharing the same agent_id.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
Fixed the cleanup-on-failure issue flagged by Greptile in 2da63c0 — wrapped `run_subagent()` body in `try/finally` to ensure `_cleanup_isolation()` always runs, even when `_create_subagent_thread()` raises.
